var/spack/repos/builtin/packages/edm4hep/package.py

```diff
@@ -88,6 +96,8 @@ def cmake_args(self):
         # C++ Standard
         args.append(self.define("CMAKE_CXX_STANDARD", self.spec.variants["cxxstd"].value))
         args.append(self.define("BUILD_TESTING", self.run_tests))
+        if self.spec.satisfies("@0.99.2: +json"):
```

Comment:
Gotcha: `nlohmann-json` is (currently) always in the build of `edm4hep`:
- `root` requires `nlohmann-json`, and as `root` is required directly and through `podio`, you always have root and all its link dependencies, including (currently) `nlohmann-json`.
- While `nlohmann-json` is a header-only library, root does not use `type="build"` for it.
  - If root is updated to use  `nlohmann-json` with  `type="build"`, you might get it removed.
  - Still, any other package in the build could pull into the build.

So, if you don't want to have `@main` to always silently be compiled without telling cmake and as a result always using the default of `EDM4HEP_WITH_JSON=ON`, you also should add it for `@main`:

```suggestion
        if self.spec.satisfies("@0.99.2:"):
```

Reply:
Doesn't `@main` compare higher than `@0.99.2:`? I always thought this was the case and at least seem to remember that this worked as expected in the past without an explicit `@main`. However, I have not tested this explicitly yet.

Reply:
> Doesn't @main compare higher than @0.99.2:? I always thought this was the case and at least seem to remember that this worked as expected in the past without an explicit @main. However, I have not tested this explicitly yet.

It's your call. I was under the impression that it didn't when I tested, but feel free to test or not. However, if you really like to disable the `json` support, because `nlohmann-json` is a dependency of `root` (and it's currently not declared to be a build-only dependency by it) in the variant `~json`, because your `depends_on()` calls are not active (by then having `, when="+json"`) you get whatever version of `nlohmann-json` spack (or the user) decides to use for compiling `root`, and that may well be outside the version range that `edm4json` supports.

That means, when you really want to have a `~json` variant to actually do that `~json` means, you have to pass `EDM4HEP_WITH_JSON=OFF`, which means that you have to remove the `+json` from this satisfies.

I am removing this additional secondary if for testing `@main`, as your observation from the past is different from what I remember or think my own testing showed now (otherwise I would not have come up with adding it).

Anyway, I'm removing it from my suggestion. But just for reference, it was:
```py
 if self.spec.satisfies("@0.99.2:") or self.spec.satisfies("@main"):
```

But if you want `~json`, I stronly suggest that you actually test that after the build, your compiled binary actually has no json support in it.